src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -20,19 +20,26 @@ internal static class BitOps
         // PR already merged: https://github.com/dotnet/coreclr/pull/22118
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static int TrailingZeroCount(int matches)
+        public static byte TrailingZeroCount(int value)
+            // Changed parameter name to something more useful
+            // Changed return type from int to byte so that caller can assign it to any size/sign of int
+            => TrailingZeroCount((uint)value);
+
+        // TODO: Consolidate with similar methods below
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static byte TrailingZeroCount(uint value)
```

Comment:
Suggested changes. See overload above.

Reply:
I think just returning `uint` is preferable. It is the natural register size and avoids the need to insert zero/sign-extensions.

Reply:
I originally used `uint` in all my own code, but then discovered that when writing consuming code I'd need to _explicitly_ up/down-convert based on the receiving variable, which may be any integer size or sign. Byte obviously _implicitly_ converts into (nearly) all of them.
But happy to make the change if you think it's a better api regardless.

Reply:
C# implicitly does all operations as `int/uint` or `long/ulong` (that is, `byte + byte = int`). So I think matching that is generally beneficial.

The user can explicitly downcast if they need to store the result anywhere.

Reply:
Done